test/jdk/java/lang/StableValue/JEP.md

```diff
@@ -0,0 +1,511 @@
+
+```
+class Cahced {
+
+    // 1. Centrally declare a cached function backed by a map of stable values
+    private static final Function<String, Logger> LOGGERS =
+            StableValue.newCachingFunction(Set.of("com.company.Foo", "com.company.Bar"),
+            Logger::getLogger, null);
+
+    private static final String NAME = "com.company.Foo";
+
+    // 2. Access the cached value via the function with as-declared-final
+    //    performance (evaluation made before the first access)
+    Logger logger = LOGGERS.apply(NAME);
+}
+```
+
+It should be noted that the enumerated set of valid inputs given at creation time constitutes the only valid
+input keys for the cached function. Providing a non-valid input for a cached `Function` (or a cached `IntFunction`)
+would incur an `IllegalArgumentException`.
+
+An advantage with cached functions, compared to working directly with `StableValue` instances, is that the
+initialization logic can be centralized and maintained in a single place, usually at the same place where
+the cached function is defined.
+
+Additional cached function types, such a cached `Predicate<K>` (backed by a lazily computed
+`Map<K, StableValue<Boolean>>`) or a cached `BiFunction` can be custom-made. An astute reader will be able
+to write such constructs in a few lines.
+
+#### Background threads
+
+As noted above, the cached-returning factories in the Stable Values & Collections API offers an optional
+tailing thread factory parameter from which new value-computing background threads will be created:
+
+```
+// 1. Centrally declare a cached function backed by a map of stable values
+//    computed in the background by two distinct virtual threads.
+private static final Function<String, Logger> LOGGERS =
+        StableValue.newCachingFunction(Set.of("com.company.Foo", "com.company.Bar"),
+        Logger::getLogger,
+        // Create cheap virtual threads for background computation
+        Thread.ofVirtual().factory());
+
+... Somewhere else in the code
+
+private static final String NAME = "com.company.Foo";
+
+// 2. Access the cached value via the function with as-declared-final
+//    performance (evaluation made before the first access)
+Logger logger = LOGGERS.apply(NAME);
+```
+
+This can provide a best-of-several-worlds situation where the cached function can be quickly defined (as no
+computation is made by the defining thread), the holder value is computed in background threads (thus neither
+interfering significantly with the critical startup path nor with future accessing threads), and the threads actually
+accessing the holder value can directly access the holder value with as-if-final performance and without having to compute
+a holder value. This is true under the assumption, that the background threads can complete computation before accessing
+threads requires a holder value. If this is not the case, at least some reduction of blocking time can be enjoyed as
```

Comment:
We should mention that if background thread is computing, accessors will block.

Reply:
Don't you think that is covered by "... reduction of blocking time ..."?

Reply:
Sure.